src/test/java/com/coherent/task60/SingletoneWebDriver.java

```diff
@@ -1,28 +1,85 @@
 package com.coherent.task60;
 
+import org.openqa.selenium.Capabilities;
 import org.openqa.selenium.Dimension;
+import org.openqa.selenium.MutableCapabilities;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.edge.EdgeDriver;
+import org.openqa.selenium.edge.EdgeOptions;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.firefox.FirefoxOptions;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
 
 public final class SingletoneWebDriver {
-    private static WebDriver driver;
+    private static RemoteWebDriver driver;
+    private static PropertiesHelper propertiesHelper = new PropertiesHelper();
+    private static String environment1PropFile = "environment_1.properties";
 
-   private SingletoneWebDriver() {
+    private SingletoneWebDriver() {
     }
 
-    public static WebDriver getDriver(){
-        ChromeOptions options = new ChromeOptions();
-        options.addArguments("--remote-allow-origins=*");
-        if(driver == null){
-            driver = new ChromeDriver(options);
+    public static WebDriver getDriver() {
+        String platformName = propertiesHelper.propertiesReader("platform", environment1PropFile);
+        String browserName = propertiesHelper.propertiesReader("browser", environment1PropFile);
+        String browserVersion = propertiesHelper.propertiesReader("browserVersion", environment1PropFile);
+        String browserArguments = propertiesHelper.propertiesReader("browser.arguments", environment1PropFile);
+        String saucelabsBuild = propertiesHelper.propertiesReader("saucelabs.build", environment1PropFile);
+        String saucelabsUrl = propertiesHelper.propertiesReader("saucelabs.url", environment1PropFile);
+        String saucelabsName = propertiesHelper.propertiesReader("saucelabs.name", environment1PropFile);
```

Comment:
Same for those 3. Looks like you don't use them

Reply:
I updated the code and now I use them all